lib/discover/global.go

```diff
@@ -179,7 +181,11 @@ func (c *globalClient) Lookup(ctx context.Context, device protocol.DeviceID) (ad
 	resp, err := c.queryClient.Get(ctx, qURL.String())
 	if err != nil {
 		l.Debugln("globalClient.Lookup", qURL, err)
-		return nil, err
+		return nil, c.connFailureBackoff(err)
+	}
+	if resp.StatusCode >= 500 {
+		l.Debugln("globalClient.Lookup", qURL, resp.Status)
+		return nil, c.connFailureBackoff(errors.New(resp.Status))
```

Comment:
Got to close the body here. Or maybe just defer close it after the error check block above, then we don't need to do it in all the different branches.